src/webring.rs

```diff
@@ -154,26 +185,32 @@ impl Webring {
         ret
     }
 
+    fn get_authority(uri: &Uri) -> Result<Intern<str>, TraverseWebringError> {
+        let authority = uri
+            .authority()
+            .ok_or_else(|| TraverseWebringError::NoAuthority(uri.to_owned()))?;
+        Intern::get_ref(authority.as_str())
+            .ok_or_else(|| TraverseWebringError::AuthorityNotFound(authority.to_owned()))
+    }
+
     fn member_idx_and_lock(
         &self,
         uri: &Uri,
-    ) -> Result<(usize, RwLockReadGuard<'_, WebringData>), TraverseWebringError> {
+    ) -> Result<(usize, Intern<str>, RwLockReadGuard<'_, WebringData>), TraverseWebringError> {
+        let interned = Self::get_authority(uri)?;
         let inner = self.inner.read().unwrap();
-        let authority = uri
-            .authority()
-            .ok_or_else(|| TraverseWebringError::NoAuthority(uri.to_owned()))?;
         Ok((
-            *inner
-                .members_table
-                .get(authority)
-                .ok_or_else(|| TraverseWebringError::AuthorityNotFound(authority.to_owned()))?,
+            *inner.members_table.get(&interned).ok_or_else(|| {
+                TraverseWebringError::AuthorityNotFound(uri.authority().unwrap().to_owned())
+            })?,
+            interned,
             inner,
         ))
     }
 
     /// Get the next page in the webring from the given URI based on the authority part
-    pub fn next_page(&self, uri: &Uri) -> Result<Arc<Uri>, TraverseWebringError> {
-        let (mut idx, inner) = self.member_idx_and_lock(uri)?;
+    pub fn next_page(&self, uri: &Uri, ip: IpAddr) -> Result<Intern<Uri>, TraverseWebringError> {
```

Comment:
Goes along with the comment on `routes.rs`, but the `ip` parameter should be optional so we can leave it out for requests with `DNT: 1`. Ditto for `(next|random)_page`.